common/remote/sockfile.cpp

```diff
@@ -4224,14 +4227,23 @@ class CRemoteIndexBaseActivity : public CRemoteDiskBaseActivity
 protected:
     bool isTlk = false;
     bool allowPreload = false;
+    unsigned crc = 0;
     Owned<IKeyIndex> keyIndex;
     Owned<IKeyManager> keyManager;
 
     void checkOpen()
     {
         if (opened)
             return;
-        keyIndex.setown(createKeyIndex(fileName, 0, isTlk, allowPreload)); // JCSMORE crc
+        Owned<IFile> indexFile = createIFile(fileName);
+        CDateTime modTime;
+        indexFile->getTime(nullptr, &modTime, nullptr);
+        time_t modTimeTT = modTime.getSimple();
+        CRC32 crc32(crc);
+        crc32.tally(sizeof(time_t), &modTimeTT);
+        crc = crc32.get();
```

Comment:
You are modifying a member variable, meaning you will get a differenc crt every time you call checkOpen. Not sure it will cause an issue but would it be cleaner for the combined crc/timestamp to be a local? Or can we guarantee that this bit of code happens only once and we WANT to update the member variable?